pkg/service/component_definition.go

```diff
@@ -222,6 +223,14 @@ func (s *service) ListComponentDefinitions(ctx context.Context, req *pb.ListComp
 		if req.GetView() != pb.ComponentDefinition_VIEW_FULL {
 			def.Spec = nil
 		}
+
+		// Temporary solution to filter out the instill-app component for local-ce:dev edition
+		// In the future, when it is a confirmed requirement, we can add a new field in the
+		// component definition to mark it as hidden.
```

Comment:
💭 In the future the non-CE definitions shouldn't be in the CE repositories and Cloud should extend the CE component store to initialise the private components.

Reply:
Yes, we think so. Thanks for pointing out. I will add it in the comment as well.